src/ganj/AIGuesser.java

```diff
@@ -0,0 +1,37 @@
+package ganj;
+
+import java.util.Random;
+import java.util.Scanner;
+
+import static ganj.CompareResult.GREATER;
+import static ganj.CompareResult.LOWER;
+
+public class AIGuesser implements Guesser {
+    private int from;
+    private int to;
+    private int guessByAI;
```

Comment:
Bad name. The class name shows you it's AI. I don't see the reason to duplicate who made this guess in a variable name. Bettter call it `lastGuess`.